### PR TITLE
fix(menu-full-screen): fix keys in rendering of children

### PR DESCRIPTION
**Pull request by robinzigmond**
Currently the component gives a warning in the console for the list of children not having keys. And due to the lack of keys, complex situations involving a child being added or removed while another changes state are not handled correctly. This has been fixed by using React's build in toArray utility which automatically handles the keys.

### Proposed behaviour

No warnings should be seen in the console when rendering the component.
No warnings should be seen in the terminal when running the unit tests.
Adding or removing child items should maintain any state in the other children.

### Current behaviour

When rendering the component, a warning about missing keys is visible in the console (this can be seen on Storybook - go to the "full screen" example and shrink the screen enough that a MenuFullScreen is rendered).
The same warning is seen in the terminal when running the unit tests for MenuFullScreen.
Adding a child item at the start of the menu resets state in the other children (see codesandbox linked below).

### Checklist

- [X] Commits follow our style guide
- [X] Related issues linked in commit messages if required
- [X] Screenshots are included in the PR if useful
- [X] All themes are supported if required
- [X] Unit tests added or updated if required
- [X] Cypress automation tests added or updated if required
- [X] Playwright automation tests added or updated if required
- [X] Storybook added or updated if required
- [X] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [X] Typescript `d.ts` file added or updated if required
- [X] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

Ensure that no console warnings about keys appear when viewing MenuFullScreen in storybook, or when running unit tests.

Retest the fix in https://github.com/Sage/carbon/pull/6155 to ensure that nothing has regressed from that (the changes in that PR are what inadvertently caused this problem with keys - note that a Cypress test added as part of that PR was unintentionally removed when reverting the Menu Cypress tests to Typescript, this has been reinstated as part of this PR).

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

https://codesandbox.io/s/dreamy-hoover-zv2z35

[explanation of Sandbox - this is a contrived example where one menu item maintains an internal state that regularly counts from 0 up to 2, while after 5 seconds a new menu item appears at the start of the menu. Observe that when the new item appears, the last item has its count reset to 0. This would not be expected to happen, and should be fixed by the build from this PR.]

### CODE REVIEW

File: src/components/menu/menu-full-screen/menu-full-screen.component.tsx [hunk 1]
```diff
@@ -77,9 +77,15 @@ export const MenuFullscreen = ({
...
-  );
+  const flattenedChildren = React.Children.toArray(children);
+  const childrenWithMenuDividers: React.ReactNode[] = [];
+  flattenedChildren.forEach((child, index) => {
+    childrenWithMenuDividers.push(child);
+    if (index < flattenedChildren.length - 1) {
+      childrenWithMenuDividers.push(<MenuDivider />);
+    }
+  });
+  const childArray = React.Children.toArray(childrenWithMenuDividers);
```
Comment (edleeks87) [inline src/components/menu/menu-full-screen/menu-full-screen.component.tsx:88]: suggestion (non-blocking): happy for you to leave this as you have it already but we could probably reduce the amount of arrays here by using `map`

```
const childArray = React.Children.toArray(flattenedChildren.map((child, index) => {
    if (index < flattenedChildren.length - 1) {
      return (
        <>
          {child}
          <MenuDivider />
        </>
      )
    }

    return child;
  }));

**Review by DipperTheDan — COMMENTED** [review]
Nothing more to add from me 😄👍🏻

File: src/components/menu/menu-full-screen/menu-full-screen.component.tsx (hunk 1 above)
Comment (robinzigmond) [reply src/components/menu/menu-full-screen/menu-full-screen.component.tsx:88]: Thanks - this seems to work just as well as far as assigning keys correctly is concerned. Happy to make this change.

**Review by DipperTheDan — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by ZhuoyuJin — APPROVED** [review]

**Pull request merged**